lib/fastlane_core/configuration/configuration_file.rb

```diff
@@ -32,10 +32,13 @@ def initialize(config, path, block_for_missing)
     def method_missing(method_sym, *arguments, &block)
       # First, check if the key is actually available
       if self.config.all_keys.include? method_sym
-        value = arguments.first || (block.call if block_given?) # this is either a block or a value
-        if value
-          self.config[method_sym] = value if self.config._values[method_sym].to_s.length == 0
-        end
+        # This silently prevents a value from having its value set more than once.
+        return unless self.config._values[method_sym].to_s.empty?
```

Comment:
I agree with the approach for this PR, add a test that it works as before but don't change two things at once.  This one is pretty easy to understand the repercussions.